### ISSUE TITLE
Need error tolerance for incorrect read structures

### ISSUE THREAD

**Issue by Shians** [LuyiTian/scPipe#44]
Currently `sc_trim_barcode` does not recognise out of bounds indices, so if we wanted to take bp10-20 from a 8bp long read we just have an empty barcode or UMI. This propagates to the `sc_demultiplex` step where empty values (or memory garbage) gets put into the tags, leading to useless bam files or even corrupt ones.

There needs to be a check and error if the read structure isn't possible within the given reads.

**Comment by Shians** [comment]
Apparently we're using memcpy so when we go out of bounds we are copying stuff from free floating memory, potentially dangerous.

**Comment by cpattaroni** [comment]
Has this issue been solved?

**Comment by Shians** [comment]
No, I don't believe we ever properly addressed this. Thanks for bringing it to my attention again, I might have some Christmas coding to do.

**Comment by cpattaroni** [comment]
I've raised an issue #129, currently trying to first filter and clean the reads both by QC and minimum length before inputting it in the `sc_trim_barcode` function.

Will let you know if it solved anything :)
